src/main/kotlin/no/nav/melosys/soknadmottak/soknad/SoknadService.kt

```diff
@@ -10,14 +12,29 @@ private val logger = KotlinLogging.logger { }
 
 @Service
 class SoknadService @Autowired constructor(
-    private val soknadRepository: SoknadRepository
+    private val soknadRepository: SoknadRepository,
+    private val dokgenConsumer: DokgenConsumer
 ) {
+    fun erSøknadArkivIkkeLagret(arkivRef: String): Boolean {
+        return soknadRepository.findByArkivReferanse(arkivRef).count() == 0
+    }
+
+    fun hentPdf(søknad: Soknad): ByteArray {
+        // TODO mapping fra søknad.innhold til soknadFelter
+        val soknadFelter = SoknadFelterBuilder().build()
+        return dokgenConsumer.lagPDF("soeknad", soknadFelter)
```

Comment:
Kommer denne applikasjonen til å ha bruk for flere forskjellige dokument/maler? I såfall kunne vi hatt f.eks. `dokgenService.lagSoeknadPDF(soknadFelter)` som holder på malnavn.

Reply:
Ja, kvitteringsmeldingen (HTML). Godt forslag 👍